src/circuit/circuit_frame.rs

```diff
@@ -836,43 +842,49 @@ fn reduce_expression<F: LurkField, CS: ConstraintSystem<F>>(
     )?;
 
     // True if make_thunk is called.
-    let make_thunk_boolean = Boolean::not(&alloc_is_zero(
+    let make_thunk_boolean = &alloc_is_zero(
         &mut cs.namespace(|| "apply_continuation_make_thunk is zero"),
         &make_thunk_num,
-    )?);
+    )?
+    .not();
 
     let thunk_results = make_thunk(
         &mut cs.namespace(|| "make_thunk"),
         &result_cont0,
         &result_expr0,
         &result_env0,
-        &make_thunk_boolean,
+        make_thunk_boolean,
         allocated_cont_witness,
         store,
         g,
     )?;
 
     let result_expr_candidate = AllocatedPtr::pick(
         &mut cs.namespace(|| "pick maybe make_thunk expr"),
-        &make_thunk_boolean,
+        make_thunk_boolean,
         &thunk_results.0,
         &result_expr0,
     )?;
 
     let result_env_candidate = AllocatedPtr::pick(
         &mut cs.namespace(|| "pick maybe make_thunk env"),
-        &make_thunk_boolean,
+        make_thunk_boolean,
         &thunk_results.1,
         &result_env0,
     )?;
 
     let result_cont_candidate = AllocatedContPtr::pick(
         &mut cs.namespace(|| "pick maybe make_thunk cont"),
-        &make_thunk_boolean,
+        make_thunk_boolean,
         &thunk_results.2,
         &result_cont0,
     )?;
 
+    // dbg!(&result_expr_candidate.fetch_and_write_str(store));
+    // dbg!(&result_env_candidate.fetch_and_write_str(store));
+    // dbg!(&result_cont_candidate.fetch_and_write_cont_str(store));
+    // dbg!(expr, env, cont);
+
```

Comment:
Do we anticipate these will be commonly needed, or can we get rid of them?

Reply:
No, it is just a rebase problem. Will remove it.